proto/grp.proto

```diff
@@ -61,6 +64,41 @@ message GetGroupResponse {
   string owned_domain = 3;
 }
 
+message GetGroupUsersRequest {
+  // Request context.
+  // This contains the group ID to get the users for.
+  context.RequestContext request_context = 1;
+
+  // Membership status filter (optional).
+  GroupMembershipStatus membership_status = 2;
+}
+
+message GetGroupUsersResponse {
+  // Users related to the group that match the request criteria.
+  repeated user_id.DisplayUser users = 1;
+}
+
+message UpdateGroupUsersRequest {
+  context.RequestContext request_context = 1;
+
+  // IDs of users whose membership should be updated.
```

Comment:
you can make this more flexible by doing:

enum GroupMembershipAction {
  UNKNOWN_MEMBERSHIP_UPDATE = 0;
  // Add the user to the group.
  ADD = 1;
  // Remove the user from the group as well as any pending membership requests.
  REMOVE = 2;
}

message UpdateGroupUsersRequest {
  context.RequestContext request_context = 1;

  GroupMemberShipUpdate {
    user_id.UserId user_id = 1;
    GroupMembershipAction action = 2;
  }
  repeated GroupMemberShipUpdate = 2;
}

Reply:
Done. Thought about doing this but wasn't sure whether the response would be "let's keep it simple for now" lol

Reply:
Went a step further and nested `GroupMembershipAction` within `GroupMembershipUpdate`. Also simplified the message names.